### ISSUE TITLE
Code not working correctly + improvements

### ISSUE THREAD

**Issue by lupusorina** [Heath-C-505/Gradient_Descent#1]
- definitely split functionality (gradient descent in one class. visualization in another class)
- capitalize the constants
- explanation of the coefficients that you used (learning rate etc)
- make the same functionality but using Pytorch implementation
- integrate the IDBD algorithm 

Good things:
- comments are good, but don't have typos
- something was working!!! yeee.
